utils/httpsignature/httpsignature.go

```diff
@@ -20,15 +21,34 @@ import (
 
 // SignatureParams contains parameters needed to create and verify signatures
 type SignatureParams struct {
-	Algorithm Algorithm
-	KeyID     string
-	Headers   []string // optional
+	Algorithm       Algorithm
+	KeyID           string
+	DigestAlgorithm *crypto.Hash // optional
+	Headers         []string     // optional
 }
 
-// Signature represents an http signature and it's parameters
-type Signature struct {
+// signature is an internal represention of an http signature and it's parameters
+type signature struct {
 	SignatureParams
-	Sig string // FIXME remove since we should always use http.Request header?
+	Sig string
+}
+
+// Signator is an interface for cryptographic signature creation
+type Signator interface {
+	Sign(rand io.Reader, digest []byte, opts crypto.SignerOpts) (signature []byte, err error)
```

Comment:
re my comment above: I suppose this `Signator` interface may need rand exposed due to signator type, but I still consider it risky.

Reply:
I probably should have an inline comment noting that this is basically a subset of the https://pkg.go.dev/crypto#Signer interface. I made the choice to use `Sign` from the standard library `crypto.Signer` rather a custom function signature a while back - by preserving it here I was trying to make sure we wouldn't  have to make changes to existing call sites

Reply:
Added a comment